ts/Extensions/RangeSelector.ts

```diff
@@ -2164,18 +2289,174 @@ class RangeSelector {
+                this.showButtons();
+            }
+        }
+    }
 
+    /**
+     * Collapse the buttons and put the select element on top.
+     *
+     * @private
+     * @function Highcharts.RangeSelector#collapseButtons
+     * @return {void}
+     */
+    public collapseButtons(): void {
+        const {
+            buttons,
+            buttonOptions,
+            dropdown,
+            zoomText
+        } = this;
+
+        const getAttribs = (text: string): SVGAttributes => ({
+            text: `${text} ▾`,
+            paddingLeft: 8,
+            paddingRight: 8,
+            width: 'auto'
+        });
+
+        if (zoomText) {
+            zoomText.hide();
+        }
+
+        let hasActiveButton = false;
+
+        buttonOptions.forEach((
+            rangeOptions: Highcharts.RangeSelectorButtonsOptions,
+            i: number
+        ): void => {
+            const button = buttons[i];
+            if (button.state !== 2) {
+                button.hide();
+            } else {
+                button.show();
+                button.attr(getAttribs(rangeOptions.text));
+
+                hasActiveButton = true;
             }
+        });
+
+        if (!hasActiveButton && buttons.length > 0) {
+            if (dropdown) {
+                dropdown.selectedIndex = -1;
+            }
+
+            buttons[0].show();
+            buttons[0].attr(getAttribs(buttonOptions[0].text));
+        }
+
+        this.positionButtons();
+        this.showDropdown();
+    }
+
+    /**
+     * @private
+     * @function Highcharts.RangeSelector#showButtons
+     * @return {void}
+     */
+    public showButtons(): void {
+        const {
+            buttons,
+            buttonOptions,
+            options,
+            zoomText
+        } = this;
+
+        this.hideDropdown();
+
+        if (zoomText) {
+            zoomText.show();
+        }
+
+        buttonOptions.forEach((
+            rangeOptions: Highcharts.RangeSelectorButtonsOptions,
+            i: number
+        ): void => {
+            buttons[i].show();
+            buttons[i].attr({
+                text: rangeOptions.text,
+                width: options.buttonTheme.width || 28,
+                paddingLeft: 0,
+                paddingRight: 0
+            });
+        });
+
+        this.positionButtons();
+    }
+
+    /**
+     * Get the index of the visible button when the buttons are collapsed.
+     *
+     * @private
+     * @function Highcharts.RangeSelector#currentButtonIndex
+     * @return {number}
+     */
+    public currentButtonIndex(): number {
+        const { dropdown } = this;
+
+        if (dropdown && dropdown.selectedIndex > -1) {
+            return dropdown.selectedIndex;
+        }
+        return 0;
+    }
+
+    /**
+     * Position the select element on top of the button.
+     *
+     * @private
+     * @function Highcharts.RangeSelector#showDropdown
+     * @return {void}
+     */
+    public showDropdown(): void {
```

Comment:
How about `positionDropdown` instead of `showDropdown`?

Reply:
Nevermind, I see now how it's hidden :)

Reply:
I think they should stay tangled, they dont make much sense on their own and then you would have 2 function calls everywhere.